cores/esp8266/spiffs_hal.cpp

```diff
@@ -122,7 +122,7 @@ int32_t spiffs_hal_write(uint32_t addr, uint32_t size, uint8_t *src) {
 
     if (alignedEnd != alignedBegin) {
         uint32_t* srcLeftover = (uint32_t*) (src + alignedBegin - addr);
-        uint32_t srcAlign = ((uint32_t) srcLeftover) & 3;
+        uint32_t srcAlign = ((uintptr_t) srcLeftover) & 3;
         if (!srcAlign) {
```

Comment:
because `uint32_t srcAlign = ((uintptr_t)... ` doesn't seem right in 64 bits and this variable is not used elsewhere, wouldn't it be cleaner to 
```
if (!(srcLeftover & 3)) {
    // source is not aligned
    ...
```
?

Reply:
I agree! I didn't realize that srcAlign was only used once. good catch!

Reply:
hmm, still does not compile, because & is not compatible to a pointer type. so the uintptr_t cast is still needed, I guess.

`if (!((uintptr_t)srcLeftover & 3)) { // source is already aligned`